pype/plugins/maya/publish/collect_instances.py

```diff
@@ -108,14 +116,36 @@ def process(self, context):
             label = "{0} ({1})".format(name,
                                        data["asset"])
 
-            if "handles" in data:
-                data["handleStart"] = data["handles"]
-                data["handleEnd"] = data["handles"]
-
             # Append start frame and end frame to label if present
             if "frameStart" and "frameEnd" in data:
-                data["frameStartHandle"] = data["frameStart"] - data["handleStart"]
-                data["frameEndHandle"] = data["frameEnd"] + data["handleEnd"]
+
+                # if frame range on maya set is the same as full shot range
+                # adjust the values to match the asset data
+                if (ctx_frame_start_handle == data["frameStart"]
+                        and ctx_frame_end_handle == data["frameEnd"]):  # noqa: W503, E501
+                    data["frameStartHandle"] = ctx_frame_start_handle
+                    data["frameEndHandle"] = ctx_frame_end_handle
+                    data["frameStart"] = ctx_frame_start
+                    data["frameEnd"] = ctx_frame_end
+                    data["handleStart"] = ctx_handle_start
+                    data["handleEnd"] = ctx_handle_end
+
+                # if there are user values on start and end frame not matching
+                # the asset, use them
+
+                else:
```

Comment:
With the optional validator in place I believe this "if" logic should be removed and only the "else" logic should remain as the main logic. @mkolar correct?

Reply:
Technically you are correct, the problem is with shots already in production. Their parameters on review sets, for example, are counting on start and end frame as being with handles in the scene and they only have the `handles` parameter. We need to avoid the situation where you open a previous file and suddenly it behaves differently than before. This mitigates that, albeit a bit magically. 

Another option, of course, is leaving the repair to the validator, however, the thought of repairing 1000 shots (one of the current production with all layout finished) is a bit gruesome. Truth is that this is needed in production asap to make sure all the publishes coming out are consistent. So we suggest merging this (after @antirotor fixes the layer overrides), but well tag this first `if statement` as `deprecated` and will remove it in pype3.0 (possibly sooner).